mmv1/products/artifactregistry/Repository.yaml

```
@@ -504,7 +514,30 @@ properties:
                 description: |-
                   Specific repository from the base, e.g. `"centos/8-stream/BaseOS/x86_64/os"`
                 immutable: true
-
+      - !ruby/object:Api::Type::NestedObject
+        name: 'upstreamCredentials'
+        description: |-
+          The credentials used to access the remote repository.
+        immutable: true
```

These fields should likely all be marked sensitive:

```suggestion
        immutable: true
        sensitive: true
```

---

if they are sensitive, I assume that the API probably doesn't return a value for them? If that's correct, you will also need to ignore the value from the API. Unfortunately ignore_read is not currently supported on nested fields: https://github.com/hashicorp/terraform-provider-google/issues/12410 You'll need a custom flatten function similar to https://github.com/GoogleCloudPlatform/magic-modules/blob/5923d4cb878396a04bed9beaf22a8478e8b1e6a5/mmv1/products/sql/SourceRepresentationInstance.yaml#L95

---

If these fields are not returned by the API, you'll also need to use `ignore_read_extra` to make them not be expected during the ImportStateVerify step.

---

The fields are returned by the API. The sensitive credentials are stored in the secret.

---

yep, I read too quickly - this is referencing a secret version for the password so all good regarding this thread, thanks!